### DISCUSSION TITLE
How to use WebSockets?

### DISCUSSION THREAD

**Discussion by ioucyf** [sveltejs/kit#8715] [Q&A]
How to set up SvelteKit server to use WebSockets with the clients?

Also, is it possible to install and utilize similar tools like socket.io with SvelteKit instead?

**Comment by geoffrich** [comment]
I haven't done it myself, but [this post](https://joyofcode.xyz/using-websockets-with-sveltekit) from @mattcroat may be helpful.

**Reply by samal-rasmussen** [reply]
It uses socket.io. Not useful for those that aren't using socket.io.

**Comment by taosx** [comment]
Are there any potential drawbacks to using a custom Express server with SvelteKit beside just not having the structure that kit offers?

**Reply by NormandoHall** [reply]
Maybe it is better to create a new thread with your question, but no, there is no problem to use a custom Express in backend. Read the docs about [Custom server](https://kit.svelte.dev/docs/adapter-node#custom-server). I am using Express as backend for all sites created with SK.

**Comment by wvhulle** [comment]
Joy of code has the tutorial

**Reply by samal-rasmussen** [reply]
He uses socket.io for his Sveltekit/WebSocket tutorial, and socket.io adds a lot of abstraction, so it isn't helpful at all for those who aren't using socket.io.
